pkg/cri/server/image_pull.go

```diff
@@ -330,6 +330,12 @@ func hostDirFromRoots(roots []string) func(string) (string, error) {
 
 // registryHosts is the registry hosts to be used by the resolver.
 func (c *criService) registryHosts(ctx context.Context, auth *runtime.AuthConfig) docker.RegistryHosts {
+	cR := c.config.Registry
+	if len(cR.Mirrors) == 0 && len(cR.Configs) == 0 && len(cR.Auths) == 0 && len(cR.Headers) == 0 {
+		// No old-style configuration, default to docker-compatible defaults
+		// See https://github.com/containerd/containerd/issues/6485
+		c.config.Registry.ConfigPath = "/etc/containerd/certs.d:/etc/docker/certs.d"
```

Comment:
This is Linux-specific. I don't know if I need to do something cross-platform here or not.

Reply:
Do you know Docker's default in Windows? @jterry75 @kevpar

Reply:
Windows server directions are here; the TL;DR is that the certificates need to be installed into the system cert store on Windows, I think: https://docs.docker.com/registry/insecure/#use-self-signed-certificates

Reply:
In addition, it seems like the Windows cert directions require restarting Docker, so it's not as seamless as the Linux process of putting certificates under the registry name in `/etc/docker/certs.d`.

Reply:
would it help to set it here https://github.com/containerd/containerd/blob/main/pkg/cri/config/config_unix.go#L29?

Reply:
would it help to set it here https://github.com/containerd/containerd/blob/main/pkg/cri/config/config_unix.go#L29?

Reply:
Sorry, needed to do a little research.

Setting the default in `config_unix` causes the default config to be _merged_ with any config in the paths, which means that you can hit the [error condition on line 400 of `config.go`](https://github.com/containerd/containerd/blob/main/pkg/cri/config/config.go#L401) by setting any of the (deprecated) `mirrors` or `tls` parameters and not explicitly setting `config_path` to the empty string.

Reply:
It is still best to catch the conflict there. This is kind of deep in the path to start applying default values. Where is it checking whether `ConfigPath` is already set?

Reply:
Good point!

I've moved this to the end of [`ValidatePluginConfig`](https://github.com/containerd/containerd/blob/main/pkg/cri/config/config.go) and updated the appropriate tests in `config_test.go`.